src/main/java/Parser.java

```diff
@@ -0,0 +1,77 @@
+package duke;
+
+/**
+ * Represents a class that deals with making sense of the user command.
+ */
+public class Parser {
+
+    Parser() {
+
+    }
+
+    /**
+     * Process the input command by the users to give the respective command to the ChatBot to execute.
+     * @param String fullCommand : the full command that is given by the users.
+     * @return command : the specific command to execute later.
+     * @throws MismatchInputException
+     * @throws TaskWithNoDescriptionException
+     */
+    Command parse(String fullCommand) throws MismatchInputException, TaskWithNoDescriptionException {
+        String[] strArr = fullCommand.split(" ");
+        String command = strArr[0];
+        if (command.equals("bye")) {
+            return new ExitCommand();
+        } else if (command.equals("list")) {
+            return new ListCommand();
+        } else if (command.equals("mark")) {
+            int taskNo = Integer.parseInt(strArr[1]);
+            return new MarkCommand(taskNo);
+        } else if (command.equals("unmark")) {
+            int taskNo = Integer.parseInt(strArr[1]);
+            return new UnmarkCommand(taskNo);
+        } else if (command.equals("todo")) {
+            String[] taskDescription = processDescription(strArr);
+            Task newTask = new ToDo(taskDescription[0]);
+            return new AddCommand(newTask);
+        } else if (command.equals("deadline")) {
+            String[] taskDescription = processDescription(strArr);
+            Task newTask = new Deadline(taskDescription[0], taskDescription[1]);
+            return new AddCommand(newTask);
+        } else if (command.equals("event")) {
+            String[] taskDescription = processDescription(strArr);
+            Task newTask = new Event(taskDescription[0], taskDescription[1]);
+            return new AddCommand(newTask);
+        } else if (command.equals("delete")) {
+            int taskNo = Integer.parseInt(strArr[1]);
+            return new DeleteCommand(taskNo);
+        } else if (command.equals("find")) {
+            String keyword = strArr[1];
+            return new FindCommand(keyword);
```

Comment:
Perhaps it would make for neater code to use an enum for the different command inputs?